platforms/core-configuration/configuration-problems-base/src/main/kotlin/org/gradle/internal/configuration/problems/ProblemsListener.kt

```diff
@@ -16,11 +16,7 @@
 
 package org.gradle.internal.configuration.problems
 
-import org.gradle.internal.service.scopes.EventScope
-import org.gradle.internal.service.scopes.Scope
 
-
-@EventScope(Scope.BuildTree::class)
```

Comment:
❓ 

For my education, does this mean we no longer annotate the service interfaces, only the implementations? I thought it was useful to have the interfaces annotated (unless implementations varied in scope)

Reply:
> ❓
> 
> For my education, does this mean we no longer annotate the service interfaces, only the implementations? I thought it was useful to have the interfaces annotated (unless implementations varied in scope)

I decided to let existing checks to guide me. Our interface-to-implementation bindings are not mandatory, it seems, and this particular interface serves two purposes: it is a service used by CC machinery, but can be also used as part of serialization and can even have some non-DI-owned implementation. I've tried to untie these two, but stopped to keep this PR focused.